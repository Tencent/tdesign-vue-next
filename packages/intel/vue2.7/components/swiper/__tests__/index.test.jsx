import { mount } from '@vue/test-utils';
import Vue from 'vue';
import Swiper, { SwiperItem } from '@/src/swiper/index.ts';

// every component needs four parts: props/events/slots/functions.
describe('Swiper', () => {
  // test props api
  describe(':props', () => {
    it('', () => {
      const wrapper = mount({
        render() {
          return (
            <Swiper>
              <SwiperItem>1</SwiperItem>
            </Swiper>
          );
        },
      });
      expect(wrapper.exists()).toBe(true);
    });
    it(':animation fade', () => {
      const wrapper = mount({
        render() {
          return (
            <Swiper animation={'fade'}>
              <SwiperItem>1</SwiperItem>
            </Swiper>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':autoplay', () => {
      const wrapper = mount({
        render() {
          return (
            <Swiper autoplay>
              <SwiperItem>1</SwiperItem>
            </Swiper>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':defaultCurrent', () => {
      const wrapper = mount({
        render() {
          return (
            <Swiper defaultCurrent={2}>
              <SwiperItem>1</SwiperItem>
              <SwiperItem>2</SwiperItem>
            </Swiper>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':direction', () => {
      const wrapper = mount({
        render() {
          return (
            <Swiper direction={'vertical'}>
              <SwiperItem>1</SwiperItem>
              <SwiperItem>2</SwiperItem>
            </Swiper>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':duration', () => {
      const wrapper = mount({
        render() {
          return (
            <Swiper duration={3000}>
              <SwiperItem>1</SwiperItem>
              <SwiperItem>2</SwiperItem>
            </Swiper>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':height', () => {
      const wrapper = mount({
        render() {
          return (
            <Swiper height={600}>
              <SwiperItem>1</SwiperItem>
              <SwiperItem>2</SwiperItem>
            </Swiper>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':interval', () => {
      const wrapper = mount({
        render() {
          return (
            <Swiper interval={3000}>
              <SwiperItem>1</SwiperItem>
              <SwiperItem>2</SwiperItem>
            </Swiper>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':loop', () => {
      const wrapper = mount({
        render() {
          return (
            <Swiper loop={true}>
              <SwiperItem>1</SwiperItem>
              <SwiperItem>2</SwiperItem>
            </Swiper>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':navigation', () => {
      const wrapper = mount({
        render() {
          return (
            <Swiper
              navigation={{
                placement: 'outside',
                showSlideBtn: 'hover',
                size: 'large',
                type: 'dots-bar',
              }}
            >
              <SwiperItem>1</SwiperItem>
              <SwiperItem>2</SwiperItem>
            </Swiper>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':theme', () => {
      const wrapper = mount({
        render() {
          return (
            <Swiper theme={'dark'}>
              <SwiperItem>1</SwiperItem>
              <SwiperItem>2</SwiperItem>
            </Swiper>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':type', () => {
      const wrapper = mount({
        render() {
          return (
            <Swiper type={'card'}>
              <SwiperItem>1</SwiperItem>
              <SwiperItem>2</SwiperItem>
            </Swiper>
          );
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  // test events
  describe('@event', () => {
    it('@change', async () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return (
            <Swiper onChange={fn}>
              <SwiperItem>1</SwiperItem>
              <SwiperItem>2</SwiperItem>
            </Swiper>
          );
        },
      });
      await Vue.nextTick();
      const swiper = wrapper.findComponent(Swiper);
      swiper.find('.t-swiper__arrow-right').trigger('click');
      await Vue.nextTick();
      // expect(swiper.emitted().change).toBeTruthy();
    });
  });

  // test additional function
  describe('additional function', () => {
    it('navigation: VNode', async () => {
      const wrapper = mount({
        render() {
          const renderNavigation = () => <div>123</div>;
          return (
            <Swiper navigation={renderNavigation()}>
              <SwiperItem>1</SwiperItem>
              <SwiperItem>2</SwiperItem>
            </Swiper>
          );
        },
      });
      await Vue.nextTick();
      expect(wrapper.element).toMatchSnapshot();
    });
    it('no valid panel', async () => {
      const wrapper = mount({
        render() {
          return <Swiper></Swiper>;
        },
      });
      await Vue.nextTick();
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
