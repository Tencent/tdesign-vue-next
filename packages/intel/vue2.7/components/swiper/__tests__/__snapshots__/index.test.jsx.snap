// Vitest Snapshot v1

exports[`Swiper > :props > :animation fade 1`] = `
<div
  class="t-swiper"
>
  <div
    class="t-swiper__wrap t-swiper--inside"
  >
    <div
      class="t-swiper__content t-swiper-fade"
    >
      <div
        class="t-swiper__container"
      />
    </div>
    <ul
      class="t-swiper__navigation t-swiper__navigation-bars"
    />
  </div>
</div>
`;

exports[`Swiper > :props > :autoplay 1`] = `
<div
  class="t-swiper"
>
  <div
    class="t-swiper__wrap t-swiper--inside"
  >
    <div
      class="t-swiper__content"
    >
      <div
        class="t-swiper__container"
        style="transform: translate3d(-0%, 0px, 0px);"
      />
    </div>
    <ul
      class="t-swiper__navigation t-swiper__navigation-bars"
    />
  </div>
</div>
`;

exports[`Swiper > :props > :defaultCurrent 1`] = `
<div
  class="t-swiper"
>
  <div
    class="t-swiper__wrap t-swiper--inside"
  >
    <div
      class="t-swiper__content"
    >
      <div
        class="t-swiper__container"
        style="transform: translate3d(-0%, 0px, 0px);"
      />
    </div>
    <ul
      class="t-swiper__navigation t-swiper__navigation-bars"
    />
  </div>
</div>
`;

exports[`Swiper > :props > :direction 1`] = `
<div
  class="t-swiper"
>
  <div
    class="t-swiper__wrap t-swiper--inside t-swiper--vertical"
  >
    <div
      class="t-swiper__content"
    >
      <div
        class="t-swiper__container"
        style="transform: translate3d(0, -0%, 0px);"
      />
    </div>
    <ul
      class="t-swiper__navigation t-swiper__navigation-bars"
    />
  </div>
</div>
`;

exports[`Swiper > :props > :duration 1`] = `
<div
  class="t-swiper"
>
  <div
    class="t-swiper__wrap t-swiper--inside"
  >
    <div
      class="t-swiper__content"
    >
      <div
        class="t-swiper__container"
        style="transform: translate3d(-0%, 0px, 0px);"
      />
    </div>
    <ul
      class="t-swiper__navigation t-swiper__navigation-bars"
    />
  </div>
</div>
`;

exports[`Swiper > :props > :height 1`] = `
<div
  class="t-swiper"
>
  <div
    class="t-swiper__wrap t-swiper--inside"
  >
    <div
      class="t-swiper__content"
    >
      <div
        class="t-swiper__container"
        style="transform: translate3d(-0%, 0px, 0px);"
      />
    </div>
    <ul
      class="t-swiper__navigation t-swiper__navigation-bars"
    />
  </div>
</div>
`;

exports[`Swiper > :props > :interval 1`] = `
<div
  class="t-swiper"
>
  <div
    class="t-swiper__wrap t-swiper--inside"
  >
    <div
      class="t-swiper__content"
    >
      <div
        class="t-swiper__container"
        style="transform: translate3d(-0%, 0px, 0px);"
      />
    </div>
    <ul
      class="t-swiper__navigation t-swiper__navigation-bars"
    />
  </div>
</div>
`;

exports[`Swiper > :props > :loop 1`] = `
<div
  class="t-swiper"
>
  <div
    class="t-swiper__wrap t-swiper--inside"
  >
    <div
      class="t-swiper__content"
    >
      <div
        class="t-swiper__container"
        style="transform: translate3d(-0%, 0px, 0px);"
      />
    </div>
    <ul
      class="t-swiper__navigation t-swiper__navigation-bars"
    />
  </div>
</div>
`;

exports[`Swiper > :props > :navigation 1`] = `
<div
  class="t-swiper"
>
  <div
    class="t-swiper__wrap t-swiper--outside t-swiper--large"
  >
    <div
      class="t-swiper__content"
    >
      <div
        class="t-swiper__container"
        style="transform: translate3d(-0%, 0px, 0px);"
      />
    </div>
    <ul
      class="t-swiper__navigation"
    />
  </div>
</div>
`;

exports[`Swiper > :props > :theme 1`] = `
<div
  class="t-swiper"
>
  <div
    class="t-swiper__wrap t-swiper--inside"
  >
    <div
      class="t-swiper__content"
    >
      <div
        class="t-swiper__container"
        style="transform: translate3d(-0%, 0px, 0px);"
      />
    </div>
    <ul
      class="t-swiper__navigation t-swiper__navigation-bars"
    />
  </div>
</div>
`;

exports[`Swiper > :props > :type 1`] = `
<div
  class="t-swiper"
>
  <div
    class="t-swiper__wrap t-swiper--inside"
  >
    <div
      class="t-swiper__content t-swiper-card"
    >
      <div
        class="t-swiper__container"
      />
    </div>
    <ul
      class="t-swiper__navigation t-swiper__navigation-bars"
    />
  </div>
</div>
`;

exports[`Swiper > additional function > navigation: VNode 1`] = `
<div
  class="t-swiper"
>
  <div
    class="t-swiper__wrap t-swiper--inside"
  >
    <div
      class="t-swiper__content"
    >
      <div
        class="t-swiper__container"
        style="transform: translate3d(-0%, 0px, 0px);"
      />
    </div>
    <div>
      123
    </div>
    <div
      class="t-swiper__arrow t-swiper__arrow--default"
    >
      <div
        class="t-swiper__arrow-left"
      >
        <svg
          class="t-icon t-icon-chevron-left"
          fill="none"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M15.91 17.5l-5.5-5.5 5.5-5.5-1.41-1.41L7.59 12l6.91 6.91 1.41-1.41z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="t-swiper__arrow-right"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M8.09 17.5l5.5-5.5-5.5-5.5L9.5 5.09 16.41 12 9.5 18.91 8.09 17.5z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Swiper > additional function > no valid panel 1`] = `
<div
  class="t-swiper"
>
  <div
    class="t-swiper__wrap t-swiper--inside"
  >
    <div
      class="t-swiper__content"
    >
      <div
        class="t-swiper__container"
        style="transform: translate3d(-0%, 0px, 0px);"
      />
    </div>
    <ul
      class="t-swiper__navigation t-swiper__navigation-bars"
    />
    <div
      class="t-swiper__arrow t-swiper__arrow--default"
    >
      <div
        class="t-swiper__arrow-left"
      >
        <svg
          class="t-icon t-icon-chevron-left"
          fill="none"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M15.91 17.5l-5.5-5.5 5.5-5.5-1.41-1.41L7.59 12l6.91 6.91 1.41-1.41z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="t-swiper__arrow-right"
      >
        <svg
          class="t-icon t-icon-chevron-right"
          fill="none"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M8.09 17.5l5.5-5.5-5.5-5.5L9.5 5.09 16.41 12 9.5 18.91 8.09 17.5z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
